### PR TITLE
fix:added mobile validation for 11 digits and on error scroll to top

### LINKED ISSUE #2675: Creating new patient - validation of the phone number is wrong
## 🐛 Bug Report

The validation of the phone number is missing,
PFA the Image

## To Reproduce

- Try creating a new patient with contact number being 11 digits
- its accepting the number

## Expected behavior

- validation should be a proper 10digit number of if ok `country code`

Screenshot 2021-05-14 at 8 44 10 PM

## Your Environment

- *node version*: 6,8,10
- *fastify version*: >=1.0.0
- *os*: Mac, Windows, Linux
- *any other relevant information*

### PR DESCRIPTION
**Pull request by jayavardhan3112**
Fixes #2675

Changes proposed in this pull request:

added mobile validation for 11 digits
scrolling to the top when there is an error while creating / updating the patient
since its misleading and user might not be aware if the error is in the button or its a validation check

### CODE REVIEW

**Comment by tehKapa** [comment]
Hi @jayavardhan3112, can you sign our Contributor License Agreement? Thanks

**Comment by jayavardhan3112** [comment]
> Hi @jayavardhan3112, can you sign our Contributor License Agreement? Thanks

I have already done that, but dont know why its not updating

**Comment by DrewGregory** [comment]
> > Hi @jayavardhan3112, can you sign our Contributor License Agreement? Thanks
> 
> I have already done that, but dont know why its not updating

I believe the reason is that the author email on your commits doesn't correspond to your GitHub email. I would recommend amending your commits so that your email corresponds to your GitHub account: https://stackoverflow.com/questions/3042437/how-to-change-the-commit-author-for-one-specific-commit

File: src/patients/util/validate-patient.ts [hunk 1]
```diff
@@ -12,7 +12,7 @@ const validateEmails = (emails: ContactInfoPiece[] | undefined) =>
 
 const validatePhoneNumbers = (phoneNumbers: ContactInfoPiece[] | undefined) =>
   (phoneNumbers ?? []).map((phone) =>
-    !validator.isMobilePhone(phone.value) ? 'patient.errors.invalidPhoneNumber' : undefined,
+  (!validator.isMobilePhone(phone.value) || !new RegExp(`^[\\s\\d]{${10}}$`).test(phone.value)) ? 'patient.errors.invalidPhoneNumber' : undefined,
```
Comment (anthonyaperez) [inline src/patients/util/validate-patient.ts:15]: Declaring a new object within a ternary operator makes it a bit hard to read...maybe we can do something like this:
```
const validPhoneRegex = /^[\\s\\d]{${10}}$/

const validatePhoneNumbers = (phoneNumbers: ContactInfoPiece[] | undefined) =>
  (phoneNumbers ?? []).map((phone) =>  
    (!validator.isMobilePhone(phone.value) || !validPhoneRegex.test(phone.value)) ? 'patient.errors.invalidPhoneNumber' : undefined,  
)
```

Another option you have is to create a static utility method to do the regex checking, but that isn't necessary here.

**Review by anthonyaperez — CHANGES_REQUESTED** [review]
Nice work! Please make sure to address the changes mentioned, including the style errors that the linter throws (you can either try `npm run lint:fix` or fix them manually).

**Pull request closed without merging**